### PR TITLE
[Enhancement][Support Struct] Support Struct subfield materialize

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

This pr avoids reading redundant data in BE by determining the struct subfield that needs to be materialized in FE.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Review by zombee0 — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/SqlToScalarOperatorTranslator.java [hunk 1]
```diff
@@ -256,31 +259,38 @@ public ScalarOperator visitSlot(SlotRef node, Context context) {
...
                 Preconditions.checkArgument(node.getUsedStructFieldPos() != null, "StructType SlotRef must have" +
                         "an non-empty usedStructFiledPos!");
                 Preconditions.checkArgument(node.getUsedStructFieldPos().size() > 0);
                 List<Integer> usedStructFieldPos = node.getUsedStructFieldPos();
-                return SubfieldOperator.build(columnRefOperator, node.getOriginType(), usedStructFieldPos);
-            } else {
-                return columnRefOperator;
+                returnValue = SubfieldOperator.build(columnRefOperator, node.getOriginType(), usedStructFieldPos);
+
+                for (int pos : node.getUsedStructFieldPos()) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/SqlToScalarOperatorTranslator.java:278]: seems this should be pushed reversely, am I getting it wrong?

Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/SqlToScalarOperatorTranslator.java]: fixed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 30 / 31 (96.77%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|0|1|00.00%|[553]|
|:large_blue_circle:|com/starrocks/analysis/SubfieldExpr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/MapType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ArrayType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/SlotDescriptor.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/SqlToScalarOperatorTranslator.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ColumnRefOperator.java|1|1|100.00%|[]|

**Review by zombee0 — APPROVED** [review]

**Review by DorianZheng — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Youngwb — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Pull request merged**